scenes/ui_elements/dialogue/components/balloon.gd

```diff
@@ -222,4 +227,15 @@ func _on_balloon_gui_input(event: InputEvent) -> void:
 func _on_responses_menu_response_selected(response: DialogueResponse) -> void:
 	next(response.next_id)
 
+
 #endregion
+
+
+func _on_spoke(_letter: String, _index: int, _speed: float) -> void:
+	if talk_sound_player.playing:
+		return
+	talk_sound_player.play()
+
+
+func _on_dialogue_finished_typing() -> void:
+	talk_sound_player.stream_paused = true
```

Comment:
I don't think you need to connect to these signals: start the stream playing in `apply_dialogue_line` in this block:

```
	if not dialogue_line.text.is_empty():
        # resume audio here
		dialogue_label.type_out()
		await dialogue_label.finished_typing
        # pause it here
```

Reply:
The reason to connect to the `spoke` signal would be if you were to also connect to the `paused_typing` signal and pause the sound during the pause. If I understand correctly the signal to indicate that dialogue has resumed in that case is `spoke`.